### PR TITLE
Fix: invalid component depends on check

### LINKED ISSUE #3635: vela-core does not report error, when component depends on invalid component
**Describe the bug**
vela-core does not report error, when component depends on invalid component

**To Reproduce**
1. apply one application, Please note that "express-server-3" depends on one invalid component "express-server-0"
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: first-vela-app
spec:
  components:
    - name: express-server-1
      type: webservice
      properties:
        image: crccheck/hello-world
        port: 8000
    - name: express-server-2
      type: webservice
      dependsOn:
       -  express-server-1
      properties:
        image: crccheck/hello-world
        port: 8080
    - name: express-server-3
      type: webservice
      dependsOn:
       -  express-server-0
      properties:
        image: crccheck/hello-world
        port: 8088
```

2. check application status
```
kubectl -n vela-system describe applications.core.oam.dev first-vela-app

```

```
Events:
  Type    Reason           Age                From         Message
  ----    ------           ----               ----         -------
  Normal  Revisioned       39s (x6 over 42s)  Application  Revisioned successfully
  Normal  PolicyGenerated  39s (x6 over 42s)  Application  Policy generated successfully
  Normal  Rendered         39s (x6 over 42s)  Application  Rendered successfully
  Normal  Parsed           38s (x7 over 42s)  Application  Parsed successfully
```

3. there is not error message reported in both "kubectl describe" and vela-core' log

**Expected behavior**
1. vela-core reports some error messages to indicate this mistake.

**Screenshots**

**KubeVela Version**

```
➜  KubeVela vela version
CLI Version: v1.3.0
Core Version: 1.3.0
GitRevision: git-3aa4412
GolangVersion: go1.17.8
➜  KubeVela
```

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by StevenLeiZhang**
Signed-off-by: StevenLeiZhang <zhangleiic@163.com>

### Description of your changes

Fixes #3635 

How to fix it?
check if the depended component is exist in all components, if not, raises one error.

report the error message in logs as below
```
I0412 10:13:34.844870       1 apply.go:345] "[Finished]: i-wnvrzjvl.apply-policies(finish apply policies)" application="vela-system/first-vela-app" controller="application" resource_version="559295" generation=1 publish_version="undefined" duration="7.3µs" spanID="i-wnvrzjvl.apply-policies"
E0412 10:13:34.845402       1 application_controller.go:194] "[handle workflow]" err="convert step: convert [apply-component]: component express-server-0 not found, which is depended by express-server-3" application="vela-system/first-vela-app" controller="application" resource_version="559295" generation=1 publish_version="undefined" spanID="i-wnvrzjvl"
I0412 10:13:34.856722       1 application_controller.go:196] "[Finished]: i-wnvrzjvl(End reconcile application)" application="vela-system/first-vela-app" controller="application" resource_version="559295" generation=1 publish_version="undefined" duration="34.8278ms" spanID="i-wnvrzjvl"

```

report error in 

```kubectl -n vela-system describe applications.core.oam.dev first-vela-app```

as below
```
  Status:               rendering
Events:
  Type     Reason           Age                    From         Message
  ----     ------           ----                   ----         -------
  Normal   Parsed           3m54s (x7 over 3m54s)  Application  Parsed successfully
  Normal   Revisioned       3m54s (x6 over 3m54s)  Application  Revisioned successfully
  Normal   PolicyGenerated  3m54s (x6 over 3m54s)  Application  Policy generated successfully
  Warning  FailedWorkflow   3m54s (x6 over 3m54s)  Application  convert step: convert [apply-component]: component express-server-0 not found, which is depended by express-server-3
➜  KubeVela
```

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/application/generator.go [hunk 1]
```diff
@@ -135,11 +143,25 @@ func convertStepProperties(step *v1beta1.WorkflowStep, app *v1beta1.Application)
...
+	if dependsOnComponentNames == nil {
+		return "", true
+	}
+	for _, dc := range dependsOnComponentNames {
+		for _, c := range allComponentNames {
+			if c == dc {
+				return "", true
+			}
+		}
+		return dc, false
```
Comment (FogDong) [inline pkg/controller/core.oam.dev/v1alpha2/application/generator.go:160]: Return here will skip the rest `dependsOnComponentNames` here.

Comment (StevenLeiZhang) [reply pkg/controller/core.oam.dev/v1alpha2/application/generator.go:160]: > Return here will skip the rest `dependsOnComponentNames` here.

Yes.
I think this check should be one pre-check before deploy the Application. If any one invalid dependency is found ,just break pre-check and report error message .

**Review by wonderflow — APPROVED** [review]
thanks

File: pkg/controller/core.oam.dev/v1alpha2/application/generator.go (hunk 1 above)
Comment (StevenLeiZhang) [reply pkg/controller/core.oam.dev/v1alpha2/application/generator.go]: > Return here will skip the rest `dependsOnComponentNames` here.

Thx.
Fixed, and add one test case for this scenario.

**Review by FogDong — APPROVED** [review]
Thanks for the contributing!

**Pull request merged**